### PR TITLE
2(4) new marshal exclusive guns

### PR DESCRIPTION
**Pull request by Iglpigl**
## About The Pull Request
Sprite screenshot:
Screenshot_1
In game screenshot of rifle(left) and it's sawn off(right):
winchester
In game screenshot of lever shotgun(left)  and it's sawn off{right):
lever shotgun

Both guns are lever(bolt) guns, one is .40 rifle that stores 6 bullets doing 54.4 damage and 24 AP, other is a shotgun that stores 9 bullets which does 64.8 damage and 9.75 AP, both are exclusively acquired from marshal uplink being 10 and 8 TC respectfully replacing STS and pug from marshal's uplink. Both can be sawn off for weaker versions which fit uniform holsters and take less space.

## Why It's Good For The Game

Marshal being +-contractor ain't fun this aims to change that a bit without a complete rework.

## Changelog
:cl:
add: Exclusively added a lever shotgun with high damage low AP, and a relatively low ammo high power .40 lever rifle to marshal's uplink, both of which can be sawn off for weaker but smaller guns that fit in holsters
imageadd: Winchester and lever shotgun and their sawn-off variants
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	force = WEAPON_FORCE_DANGEROUS
+	armor_penetration = ARMOR_PEN_GRAZING
+	caliber = CAL_SHOTGUN
+	damage_multiplier = 1.2
+	penetration_multiplier = 0.65
+	recoil_buildup = 10
+	init_offset = 0
+	max_shells = 9
+	zoom_factor = 0
+	magazine_type = /obj/item/ammo_magazine/magnum
```
Comment (thedarkcrusader) [inline code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm:53]: shotgun  is taking magnum rounds looks like

**Comment by zombiemimic** [comment]
what sort of damage do these do?

**Comment by Iglpigl** [comment]
> what sort of damage do these do?

Rifle is 1.5 DMG/1.5 AP
Shotgun is 1.2 DMG/ 0.65 AP

**Comment by zombiemimic** [comment]
> Rifle is 1.5 DMG/1.5 AP Shotgun is 1.2 DMG/ 0.65 AP

I was talking more about the actual values on a human

**Comment by Iglpigl** [comment]
Rifle is 51
Shotgun is 64.8

**Comment by Humonitarian** [comment]
Honestly no idea why we want marshall to have a choice between two almost identical guns.

The .40 lever action is a cool idea, but the shotgun seems kind of gun bloat next to it?

**Comment by Humonitarian** [comment]
> Additionally, you should give both guns a damage mod of 1 when sawn off. Far as I could tell you forgot to apply one to the .40 rifle and the shotgun having a 0.7 damage mod when it usually has a 1.2 is horrifically bad.

.40 follows the obrez sawn off

Which means recoil skyrockets, penetration and bullet speed drops, but damage is unmodified

**Comment by thedarkcrusader** [comment]
So what would you suggest to make the shotgun different from the  other Winchester rifle

**Comment by SirRichardFrancis** [comment]
image

**Review by SirRichardFrancis — APPROVED** [review]

**Comment by thedarkcrusader** [comment]
> image

image
but he approved it.

File: code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm [hunk 2]
```diff
@@ -0,0 +1,75 @@
+/obj/item/gun/projectile/boltgun/levergun
+	name = "FS BR .40 \"Svengali\""
+	desc = "A well-made Frozen Star battle rifle. Uses .40 rounds."
+	icon_state = "lever_winchester"
+	item_suffix  = "_winchester"
+	force = WEAPON_FORCE_DANGEROUS
+	armor_penetration = ARMOR_PEN_GRAZING
+	caliber = CAL_MAGNUM
+	damage_multiplier = 1.6
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm:9]: I kinda think these values are extreme, but I support Humon's point on this - we could make the rifle non-marshal only

File: code/modules/projectiles/guns/projectile/battle_rifle/boltgun.dm [hunk 3]
```diff
@@ -33,6 +33,7 @@
 	sawn = /obj/item/gun/projectile/boltgun/obrez
 	var/bolt_open = 0
 	var/item_suffix = ""
+	var/message = "bolt"
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/battle_rifle/boltgun.dm:36]:
```suggestion
	var/message = "bolt"        // what message appears when cocking, eg "You work the [bolt] open, ejecting a casing!"
```

File: code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	init_offset = 0
+	max_shells = 9
+	zoom_factor = 0
+	ammo_type = /obj/item/ammo_casing/shotgun
+	matter = list(MATERIAL_STEEL = 15, MATERIAL_PLASTIC = 10, MATERIAL_PLASTEEL = 15)
+	wielded_item_state = "_doble_shotgun"
+	sharp = FALSE
+	spawn_blacklisted = TRUE
+	sawn = /obj/item/gun/projectile/boltgun/levergun/shotgun/sawn
+	message = "lever"
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm:59]: this is inherited from parent object

File: code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	item_suffix  = "_shotgun"
+	force = WEAPON_FORCE_DANGEROUS
+	armor_penetration = ARMOR_PEN_GRAZING
+	caliber = CAL_SHOTGUN
+	damage_multiplier = 1.2
+	penetration_multiplier = 0.65
+	recoil_buildup = 10
+	init_offset = 0
+	max_shells = 9
+	zoom_factor = 0
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm:52]: this is inherited from parent object

File: code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+	ammo_type = /obj/item/ammo_casing/shotgun
+	matter = list(MATERIAL_STEEL = 15, MATERIAL_PLASTIC = 10, MATERIAL_PLASTEEL = 15)
+	wielded_item_state = "_doble_shotgun"
+	sharp = FALSE
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm:56]: this is inherited from parent object

File: code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+	spawn_blacklisted = TRUE
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm:57]: this is inherited from parent object

File: code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+	saw_off = TRUE
+	can_dual = TRUE
+	twohanded = FALSE
+
+/obj/item/gun/projectile/boltgun/levergun/shotgun
+	name = "FS BR \"Sogekihei\""
+	desc = "A well-made Frozen Star shotgun. Uses shotgun shells."
+	icon_state = "lever_shotgun"
+	item_suffix  = "_shotgun"
+	force = WEAPON_FORCE_DANGEROUS
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm:44]: this is inherited from parent object but can probably be kept for the sake of consistency

**Review by Firefox13 — COMMENTED** [review]
took you long enough

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 9]
```diff
@@ -214,6 +214,25 @@
...
+	desc = "Winchester rifle kit"
+	icon_state = "box_of_doom_big"
+	w_class = ITEM_SIZE_HUGE
+
+/obj/item/storage/box/syndie_kit/winchester/populate_contents()
+	new /obj/item/gun/projectile/boltgun/levergun(src)
+	new /obj/item/ammo_magazine/slmagnum(src)
+
+/obj/item/storage/box/syndie_kit/lshotgun
+	name = "Lever shotgun box"
```
[the remark this answers is not in the document]
Comment (Firefox13) [reply code/game/objects/items/weapons/storage/uplink_kits.dm:228]: these lower-cased names are probably gonna stick out like a sore thumb given all other kits are uppercased but this just hurts

File: code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm (hunk 4 above)
Comment (Humonitarian) [reply code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm:59]: Only if parent is tosshin, right now it is mosin

**Comment by ValoTheValo** [comment]
more IH guns, just what we needed

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 10]
```diff
@@ -214,6 +214,25 @@
 	new /obj/item/gun/projectile/automatic/sts35(src)
 	new /obj/item/ammo_magazine/lrifle(src)
 
+/obj/item/storage/box/syndie_kit/winchester
+	name = "Winchester rifle box"
+	desc = "Winchester rifle kit"
+	icon_state = "box_of_doom_big"
+	w_class = ITEM_SIZE_HUGE
```
[the remark this answers is not in the document]
Comment (Iglpigl) [reply code/game/objects/items/weapons/storage/uplink_kits.dm:221]: No, not going to change already established name, this is same as asking me to change all doble to double.

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 11]
```diff
@@ -214,6 +214,25 @@
...
+
+/obj/item/storage/box/syndie_kit/winchester/populate_contents()
+	new /obj/item/gun/projectile/boltgun/levergun(src)
+	new /obj/item/ammo_magazine/slmagnum(src)
+
+/obj/item/storage/box/syndie_kit/lshotgun
+	name = "Lever shotgun box"
+	desc = "Lever shotgun kit"
+	icon_state = "box_of_doom_big"
```
[the remark this answers is not in the document]
Comment (Iglpigl) [reply code/game/objects/items/weapons/storage/uplink_kits.dm:230]: Ditto.

**Comment by thedarkcrusader** [comment]
> more IH guns, just what we needed

Not for IH for marshals

File: code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm (hunk 2 above)
Comment (Iglpigl) [reply code/modules/projectiles/guns/projectile/battle_rifle/levergun.dm:9]: Values are tested and working correctly, they are alright and considering we don't have sprites for this guns gun parts I will have to disagree.

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 12]
```diff
@@ -43,11 +43,24 @@
...
+	item_cost = 10
+	antag_roles = list(ROLE_MARSHAL)
+	path = /obj/item/storage/box/syndie_kit/winchester
+
+/datum/uplink_item/item/visible_weapons/lshotgun
+	name = "FS BR \"Sogekihei\""
+	item_cost = 8
+	antag_roles = list(ROLE_MARSHAL)
+	path = /obj/item/storage/box/syndie_kit/lshotgun
 /datum/uplink_item/item/visible_weapons/pug
```
Comment (Shadow-Quill) [inline code/datums/uplink/highly visible and dangerous weapons.dm:60]:
```suggestion
	path = /obj/item/storage/box/syndie_kit/lshotgun

 /datum/uplink_item/item/visible_weapons/pug
```
Make sure to separate code chunks (out whatever the actual name is)

File: code/game/objects/items/weapons/storage/uplink_kits.dm [hunk 13]
```diff
@@ -214,6 +214,25 @@
...
+/obj/item/storage/box/syndie_kit/lshotgun
+	name = "lever shotgun box"
+	desc = "lever-action shotgun kit"
+	icon_state = "box_of_doom_big"
+	w_class = ITEM_SIZE_HUGE
+
+/obj/item/storage/box/syndie_kit/lshotgun/populate_contents()
+	new /obj/item/gun/projectile/boltgun/levergun/shotgun(src)
+	new /obj/item/ammo_casing/shotgun/prespawned(src)
 /obj/item/storage/box/syndie_kit/pug
```
Comment (Shadow-Quill) [inline code/game/objects/items/weapons/storage/uplink_kits.dm:236]:
```suggestion
	new /obj/item/ammo_casing/shotgun/prespawned(src)

 /obj/item/storage/box/syndie_kit/pug
```
Spacing

**Review by Shadow-Quill — CHANGES_REQUESTED** [review]

**Review by Shadow-Quill — APPROVED** [review]
Code looks good.

**Comment by Rexdyer1** [comment]
I'm a fan of this honestly. we really need to move from using traitor uplinks for marshalls...maybe someday we can have a specialized background for their uplink and tactical gear. maybe being able to buy ironhammer hardsuits as well....anyways I really like these new cowboy guns, very cool looking

**Comment by Humonitarian** [comment]
Statwise the two currently are not... Very equal, but not the way the TC implies.

When taking 50 armor as a base, the .40 will cause marginally more agony, but overall **less** damage than the shotgun, while having a smaller clip.
The higher style damage of .40 helps give it an edge, but not probably by enough.

At low armour levels, the shotgun is butter-clear superior in damage.

Overall ironic, how the .40 is good against high style high armour, while the shotgun is good against low style low armour. They kind of play against eachother, being effectively identical choices with the shotgun being slightly superior until HV ammo, recoil or serb-level armor are brought into question.

**Comment by Iglpigl** [comment]
> Statwise the two currently are not... Very equal, but not the way the TC implies.
> 
> When taking 50 armor as a base, the .40 will cause marginally more agony, but overall **less** damage than the shotgun, while having a smaller clip. The higher style damage of .40 helps give it an edge, but not probably by enough.
> 
> At low armour levels, the shotgun is butter-clear superior in damage.
> 
> Overall ironic, how the .40 is good against high style high armour, while the shotgun is good against low style low armour. They kind of play against eachother, being effectively identical choices with the shotgun being slightly superior until HV ammo, recoil or serb-level armor are brought into question.

https://www.artstation.com/artwork/8eG5Wm

**Comment by Humonitarian** [comment]
> > Statwise the two currently are not... Very equal, but not the way the TC implies.
> > When taking 50 armor as a base, the .40 will cause marginally more agony, but overall **less** damage than the shotgun, while having a smaller clip. The higher style damage of .40 helps give it an edge, but not probably by enough.
> > At low armour levels, the shotgun is butter-clear superior in damage.
> > Overall ironic, how the .40 is good against high style high armour, while the shotgun is good against low style low armour. They kind of play against eachother, being effectively identical choices with the shotgun being slightly superior until HV ammo, recoil or serb-level armor are brought into question.
> 
> https://www.artstation.com/artwork/8eG5Wm

:society_cat:

**Comment by Voldirs** [comment]
Unblacklist them pls, antag guns should appear in loot table so there is no unambiguous way to identify marshall using those guns

**Review by Firefox13 — APPROVED** [review]

**Comment by Firefox13** [comment]
> Unblacklist them pls, antag guns should appear in loot table so there is no unambiguous way to identify marshall using those guns

No

**Comment by Voldirs** [comment]
> No

bruh

**Pull request merged**